src/nft_messages.rs

```diff
@@ -1,37 +1,59 @@
 use gstd::{msg, prelude::*, ActorId};
+use market_io::{ContractId, TokenId, TransactionId};
+use nft_io::*;
 use primitive_types::U256;
+
 pub type Payout = BTreeMap<ActorId, u128>;
-use nft_io::*;
 
 pub async fn nft_transfer(
+    transaction_id: TransactionId,
     nft_program_id: &ActorId,
     to: &ActorId,
     token_id: U256,
-    amount: u128,
-) -> Payout {
-    let response: NFTEvent = msg::send_for_reply_as(
+) -> Result<(), ()> {
+    msg::send_for_reply_as::<NFTAction, NFTEvent>(
         *nft_program_id,
-        NFTAction::TransferPayout {
+        NFTAction::Transfer {
+            transaction_id,
             to: *to,
             token_id,
+        },
+        0,
+    )
+    .expect("Error in sending a message `NFTAction::Transfer`")
+    .await
+    .map(|_| ())
+    .map_err(|_| ())
```

Comment:
```suggestion
```
Why?

Reply:
I think this is 'idiomatic' way to cast return value in this case. Since we return `Result<(), ()>`, matching arms should contain only unit types. @shamilsan could explain better i think.

Reply:
But we can just return `bool` as an alternative 😄